pkg/iter/iterator.go

```diff
@@ -503,11 +503,18 @@ func (i *timeRangedIterator) Next() bool {
 	ts := i.EntryIterator.Entry().Timestamp
 	for ok && i.mint.After(ts) {
 		ok = i.EntryIterator.Next()
+		if !ok {
+			continue
+		}
 		ts = i.EntryIterator.Entry().Timestamp
 	}
-
-	if ok && (i.maxt.Before(ts) || i.maxt.Equal(ts)) { // The maxt is exclusive.
-		ok = false
+	if ok {
```

Comment:
couldn't this more simply be expressed as 
```go
	if i.maxt.Before(ts) || i.maxt.Equal(ts) {
		ok = false
	}
```

Reply:
No this is a very specific path covering the case where both maxt == mint == ts.

Reply:
The shortcut is required. I tried it but the test wasn't passing.